nativelib/src/main/scala/scala/scalanative/native/pure.scala

```diff
@@ -0,0 +1,10 @@
+package scala.scalanative
+package native
+
+/** Let optimizer assume that given method is pure.
+  * This means that repetitive calls to the same method
+  * with the same arguments are allowed to be eliminated.
+  * Additionally, if method result to given method is not used
+  * then the call can be elided altogether.
+  */
+final class pure extends scala.annotation.StaticAnnotation
```

Comment:
Lol! [Here I am](https://github.com/scala-js/scala-js/issues/2227) defending myself against adding such an annotation in Scala.js, and Scala Native already has it ...